contrib/buildrpm/buildrpmLibfabric.sh

```diff
@@ -0,0 +1,368 @@
+###################
+usage="Usage: $0 [-i provider_name] [-e provider_name]
+       [-n] [-o] [-m] [-d] [-u] [-s] [-v] [-h] tarball
+
+ Provider options:
+
+  -i    provider_name
+    include 'provider_name' provider support to the build
+
+  -e    provider_name
+    exclude 'provider_name' provider support from the build
+
+ General options:
+  -n         no op, do nothing
+               useful with -v option
+
+  -o         install under /opt/libfabric/_VERSION_
+               {default: install under /usr/ }
+
+  -m         create modulefile
+               {default: don't create modulefile}
+
+  -d         build with Debugging support
+              {default: without debugging support}
+
+  -u         build with usNIC support only (do NOT combine with other build options)
+               {default: build with support for usNIC, Infiniband, Sockets, etc.}
+
+  -s         try to unpack libfabric.spec file from tarball
+               {default: look for libfabric.spec file in current directory}
+
+  -v         be verbose
+
+  -h         print this message and exit
+
+  tarball    path to Libfabric source tarball
+  "
+
+############
+# parse args
+############
+export arguments="$@"
+while getopts nomi:e:udsvh flag; do
+    case "$flag" in
+      n) noop="true"
+         ;;
+      o) install_in_opt="true"
+         ;;
+      m) create_modulefile="true"
+         ;;
+      i) configure_options="$configure_options --enable-$OPTARG"
+         ;;
+      e) configure_options="$configure_options --disable-$OPTARG"
+         ;;
+      u) configure_options="--enable-usnic --disable-sockets --disable-verbs --disable-psm"
+         ;;
+      d) configure_options="$configure_options --enable-debug"
+         ;;
+      s) unpack_spec="true"
+         ;;
+      v) verbose="true"
+         ;;
+      h) echo "$usage"
+         exit 0
+         ;;
+    esac
+done
+shift $(( OPTIND - 1 ));
+
+########################################
+# Check if there is at least 1 parameter
+# specified (tarball is mandatory)
+########################################
+if [[ $# -lt 1 ]]; then
+  echo "$usage" 1>&2
+  exit 1
+fi
+
+#############
+# Print hello
+#############
+if [[ -n "$verbose" ]]; then
+cat <<EOF
+
+------------------------------------------------------------------------------
+====              Welcome to Libfabric RPM build script                   ====
+------------------------------------------------------------------------------
+You have specified this options:
+ $arguments
+------------------------------------------------------------------------------
+
+EOF
+  if [[ -n "$noop" ]]; then
+    echo "$0: --> Option -n was specified. I won't do anything."
+  fi
+fi
+
+################################
+# Check if tarball was specified
+################################
+export tarball="$1"
+
+if [[ ! -f "$tarball" ]]; then
+  echo "$0: FATAL: Can't find $tarball" 1>&2
+  exit 2
+fi
+if [[ -n "$verbose" ]]; then
+  echo "$0: --> Found tarball: $tarball"
+fi
```

Comment:
FWIW, I sometimes like to have a `verbose` subroutine for stuff like this.  That allows you to not have to have an `if` block for every statement that you want to conditionally display.  For example:

```sh
verbose() {
    if test $verbose -eq 1; then
        echo "$0: --> $*"
    fi
}

verbose "Found tarball: $tarball"
```